docs/dockermachine.md

```diff
@@ -0,0 +1,449 @@
+- with existing automation technologies, which usually entail a quite high skill 
+threshold
+
+Docker's [`machine`](https://github.com/docker/machine) is a tool for making the 
+process of creating and managing those machines (and running Docker commands 
+against them) much faster and easier for users.  `machine` allows users to 
+quickly create running instances of the Docker daemon on local virtualization 
+platforms (e.g. Virtualbox) or on cloud providers (e.g. AWS EC2) that they can 
+connect to and control from their local Docker client binary.
+
+## Installation
+
+Docker Machine is supported on Windows, OSX, and Linux.  To install Docker 
+Machine, download the appropriate binary for your OS and architecture to the 
+correct place in your `PATH`:
+
+- [Windows - x86_64]()
+- [OSX - x86_64]()
+- [Linux - x86_64]()
+- [Windows - i386]()
+- [OSX - i386]()
+- [Linux - i386]()
+
+Now you should be able to check the version with `machine -v`:
+
+```
+$ machine -v
+machine version 0.0.2
+```
+
+## Getting started with Docker Machine using a local VM
+
+Let's take a look at using `machine` to creating, using, and managing a Docker 
+host inside of [VirtualBox](ihttps://www.virtualbox.org/).
+
+First, ensure that 
+[VirtualBox 4.3.20](https://www.virtualbox.org/wiki/Downloads) is correctly 
+installed on your system. 
+
+If you run the `machine ls` command to show all available machines, you will see 
+that none have been created so far.
+
+```
+$ machine ls
+NAME   ACTIVE   DRIVER   STATE   URL
+```
+
+To create one, we run the `machine create` command, passing the string 
+`virtualbox` to the `--driver` flag.  The final argument we pass is the name of 
+the machine - in this case, we will name our machine "dev".
+
+This will download a lightweight Linux distribution 
+([boot2docker](https://github.com/boot2docker/boot2docker)) with the Docker 
+daemon installed, and will create and start a VirtualBox VM with Docker running.
+
+
+```
+$ machine create --driver virtualbox dev
+INFO[0000] Creating SSH key...
+INFO[0000] Creating VirtualBox VM...
+INFO[0007] Starting VirtualBox VM...
+INFO[0007] Waiting for VM to start...
+INFO[0038] "dev" has been created and is now the active machine. To point Docker at this machine, run: export DOCKER_HOST=$(machine url) DOCKER_AUTH=identity
+```
+
+You can see the machine you have created by running the `machine ls` command 
+again:
+
+```
+$ machine ls
+NAME      ACTIVE   DRIVER       STATE     URL
+dev       *        virtualbox   Running   tcp://192.168.99.100:2375
+```
+
+The `*` next to `dev` indicates that it is the active host.
+
+Next, as noted in the output of the `machine create` command, we have to tell 
+Docker to talk to that machine directly by setting the `DOCKER_HOST` 
+and `DOCKER_AUTH` environment variables:
+
+```
+$ export DOCKER_HOST=$(machine url) DOCKER_AUTH=identity
+```
+
+You can now run Docker commands on this host:
+
+```
```

Comment:
Ah yes.  All bets are off if a `machine create` fails.  A change just got merged in to automatically remove machines if their creation fails, so hopefully issues like this shouldn't arise again.